src/deepsparse/pipeline.py

```diff
@@ -176,29 +177,34 @@ def __init__(
 
         self._batch_size = self._batch_size or 1
 
-    def __call__(self, *args, **kwargs) -> BaseModel:
+    def __call__(self, monitoring: bool = False, *args, **kwargs) -> BaseModel:
         if "engine_inputs" in kwargs:
             raise ValueError(
                 "invalid kwarg engine_inputs. engine inputs determined "
                 f"by {self.__class__.__qualname__}.parse_inputs"
             )
+        timer = TimingBuilder()
+        timer.initialize()
 
         # parse inputs into input_schema
+        timer.start("total_inference")
```

Comment:
Can we move all these strings to global constants so it'll be easy to change if necessary?

```
TOTAL_INFERENCE = "total_inference"
...
timer.start(TOTAL_INFERENCE)
```

Wanted to do that with all the sample_inputs/sample-inputs too 😁